app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt

```diff
@@ -444,6 +445,45 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
         return (actualNumberOfUsers * 100f) / totalNumberOfUsers
     }
 
+    /**
+     * Set up the feedback section
+     *
+     * If the user is logged in, show the post feedback section otherwise display an appropriate message
+     */
+    private fun setFeedback() {
+        if (PrefManager.getToken() != null) {
+            tvPostFeedbackDesc.visibility = View.VISIBLE
+            layoutPostFeedback.visibility = View.VISIBLE
+            buttonPost.setOnClickListener {
+                if (etFeedback.text.toString().isNotEmpty()) {
+                    val obj = PostFeedback()
```

Comment:
Better naming

Reply:
Also, all logic should be inside presenter

Reply:
@iamareebjamal This is the code after changing the POJO to data class. Here, I am simply checking for an empty String and then calling the postFeedback() implemented in the Skill Details Presenter. And, in case the String is empty I am displaying a Toast. Toast would be displayed in the view. And, the logic for posting feedback has been defined in the presenter already. I am simply sending data though and object from the view to the presenter.
Do I need to write even these lines inside the Presenter layer or is it fine?
```
if (etFeedback.text.toString().isNotEmpty()) {
                    val queryObject = PostFeedback(skillData.model, skillData.group, skillData.language,
                            skillData.skillTag, etFeedback.text.toString(), PrefManager.getToken().toString())

                    skillDetailsPresenter.postFeedback(queryObject)
                } else {
                    Toast.makeText(context, getString(R.string.toast_empty_feedback), Toast.LENGTH_SHORT).show()
                }
```